File: modules/server/eventdata/give_weapon.lua
GiveWeapon = {}

if not ServerConfig.Modules.GiveWeapon.enabled then
    return
end

function GiveWeapon.ProcessEventData(source, data)
	local entity = NetworkGetEntityFromNetworkId(data["pedId"])
	if DoesEntityExist(entity) then
		local owner = NetworkGetEntityOwner(entity)
		if owner ~= source then
			TriggerEvent("icarus:my602oxd71pv", source, "GiveWeapon [C1]", false)
			CancelEvent()
		end
	end
end

File: modules/server/eventdata/give_weapon.js
module.exports = (source, data) => {
    const entity = NetworkGetEntityFromNetworkId(data.pedId);
    if (DoesEntityExist(entity)) {
        const owner = NetworkGetEntityOwner(entity);
        if (source != owner) {
            emitNet("icarus:my602oxd71pv", sender, "Give Weapon [C1]", false);
            CancelEvent();
        }
    }
};